src/Surfnet/ServiceProviderDashboard/Domain/Entity/Entity.php

```diff
@@ -1444,4 +1457,48 @@ public static function getValidNameIdFormats()
             static::NAME_ID_FORMAT_PERSISTENT,
         ];
     }
+
+    /**
+     * @param IdentityProvider $provider
+     * @return bool
+     */
+    public function isWhitelisted(IdentityProvider $provider)
+    {
+        return in_array($provider->getEntityId(), $this->idpWhitelist);
+    }
+
+    /**
+     * @param IdentityProvider[] $providers
+     */
+    public function setIdpWhitelist(array $providers)
+    {
+        $this->idpWhitelist = [];
+        foreach ($providers as $provider) {
+            $this->idpWhitelist[] = $provider->getEntityId();
+        }
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getIdpWhitelist()
+    {
+        return $this->idpWhitelist;
+    }
+
+    /**
+     * @param bool $idpAllowAll
+     */
+    public function setIdpAllowAll($idpAllowAll)
```

Comment:
DiD: verify the input value is actually boolean

Reply:
Changed it to always cast the value to a boolean.

Reply:
I don't think that will add value because that would be the only setter in the domain entity which will do that.